noxfile.py

```diff
@@ -0,0 +1,261 @@
+"""
+Nox is a command-line tool that automates testing in multiple Python environments, similar to tox.
+Unlike tox, Nox uses a standard Python file for configuration.
+
+To run all sessions defined in the noxfile.py file, run the following command:
+$ nox
+
+To run a specific session, run the following command:
+$ nox -s <session-name>
+
+To run a session with additional arguments, run the following command:
+$ nox -s <session-name> -- <additional-arguments>
+
+To list all available sessions, run the following command:
+$ nox -l
+"""
+
+import nox
+from nox.sessions import Session
+
+# default sessions (sorted alphabetically)
+nox.options.sessions = ["bandit", "black", "flake8", "isort", "mypy", "unit-tests"]
```

Comment:
I don't know. That would mean different behavior compared to GH actions.

Reply:
> My reasoning is that it is nice to fail as fast as possible when working on a change locally.

I understand the sentiment and also see some value in this, but then

>  if there is a flake8 issue I know that I need to fix it or it will cascade to other linters, it does not make sense to wait until all targets are done

This isn't true for black/isort errors, so e.g. what happens for me often is that I have flake8 + formatter errors reported and currently all it takes is one execution of tox to tell me about all of the errors immediately and I don't need to run tox more than twice per reported error (1 initial and 1 to verify). Linters are pretty fast IMO, most time is consumed by mypy and UTs which this setting would not solve. Also, I think the overall team sentiment is to move off of these individual linters onto Ruff which will just render all this irrelevant given that it's written in a compiled language and so, again, the only real bottleneck in nox execution will be the UTs.

Reply:
The same holds true for UTs: if py3.9 fails then there is no need to run other targets.

Reply:
UTs take most of the execution time, so if you change the order in which targets are executed by stopping there at failing UTs you're deliberately robbing yourself of the opportunity to fix everything in one go without having to re-run the sessions more twice, so it's IMO actually worse that way (I mean run UTs first and skip linters on failure)